resources/default.rb

```diff
@@ -1,18 +1,170 @@
-actions :install, :uninstall
-default_action :install
-
-attribute :servicename, kind_of: String, name_attribute: true
-attribute :user, kind_of: [String, NilClass]
-attribute :group, kind_of: [String, NilClass]
-attribute :password, kind_of: [String, NilClass], sensitive: true
-attribute :url, kind_of: String, default: lazy { node['nexus3']['url'] }
-attribute :checksum, kind_of: String, default: lazy { node['nexus3']['checksum'] }
-attribute :path, kind_of: String, default: lazy { node['nexus3']['path'] }
-attribute :data, kind_of: String, default: lazy { node['nexus3']['data'] }
-attribute :home, kind_of: String, default: lazy { node['nexus3']['home'] }
-attribute :properties_cookbook, kind_of: String, default: lazy { node['nexus3']['properties_cookbook'] }
-attribute :properties_source, kind_of: String, default: lazy { node['nexus3']['properties_source'] }
-attribute :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
-attribute :vmoptions_cookbook, kind_of: String, default: lazy { node['nexus3']['vmoptions_cookbook'] }
-attribute :vmoptions_source, kind_of: String, default: lazy { node['nexus3']['vmoptions_source'] }
-attribute :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+property :instance_name, kind_of: String, name_attribute: true
+property :nexus3_user, kind_of: [String, NilClass], default: lazy { node['nexus3']['user'] }
+property :nexus3_group, kind_of: [String, NilClass], default: lazy { node['nexus3']['group'] }
+# property :nexus3_password, kind_of: [String, NilClass], sensitive: true # Admin password
+property :version, kind_of: String, default: lazy { node['nexus3']['version'] }
+property :url, kind_of: String, default: lazy { node['nexus3']['url'] }
+property :checksum, kind_of: String, default: lazy { node['nexus3']['checksum'] }
+property :nexus3_home, kind_of: String, default: lazy { node['nexus3']['home'] }
+property :path, kind_of: String, default: lazy { node['nexus3']['path'] }
+property :data, kind_of: String, default: lazy { node['nexus3']['data'] }
+property :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
+property :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+
+load_current_value do
+  # TODO: check if user, group, etc. exist for each property or so. But: do we need to do so?
+end
+
+# TODO: use helpers to make this section smaller?
+action :install do
+  install_dir = ::File.join(path, "nexus-#{version}")
+
+  user new_resource.nexus3_user do
+    comment 'Nexus 3 user'
+    home nexus3_home
+    manage_home false # is linked to install_dir below
+    shell '/bin/bash'
+  end
+
+  group new_resource.nexus3_group do
+    members nexus3_user
+    append true
+  end
+
+  # Install Nexus3 software
+  include_recipe 'ark'
```

Comment:
And have to make advice people to include the recipe default when using the resource (I think the only problem is on Windows, Linux just install a package so we can just include the package but let's talk about it if you want)